model/src/main/kotlin/Repository.kt

```diff
@@ -31,48 +31,51 @@ data class Repository(
     /**
      * Original VCS-related information from the working tree containing the analyzer root.
      */
-    val vcs: VcsInfo,
-
-    /**
-     * Processed VCS-related information from the working tree containing the analyzer root that has e.g. common
-     * mistakes corrected.
-     */
-    val vcsProcessed: VcsInfo = vcs.normalize(),
+    val provenance: KnownProvenance,
 
     /**
      * A map of nested repositories, for example Git submodules or Git-Repo modules. The key is the path to the
      * nested repository relative to the root of the main repository.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val nestedRepositories: Map<String, VcsInfo> = emptyMap(),
+    val nestedRepositories: Map<String, KnownProvenance> = emptyMap(),
```

Comment:
Nested repositories can remain VcsInfo or alternatively repository provenance, because by definition this can only be a repository.

Reply:
So Provenances have no hierarchy? The information required to create `nestedRepositories` is only available when using Git as a VCS?

Reply:
Correct, `Git` and `GitRepo` are the only implementors of `WorkingTree.getNested()`.

Reply:
I have now reverted the `nestedRepositories` changes (https://github.com/oss-review-toolkit/ort/commit/ac06be18706675938b99da8a6b3a2e1f8173551e) and will squash them into the prior commit (https://github.com/oss-review-toolkit/ort/commit/665ab205ade0560c97e1a4929b07398c1d8fc484) once all checks pass.

Rebasing on current main was also no issue, so I guess I will be done soon.

Reply:
Just pushed the rebased and squashed version.